cmd/lotus-chainwatch/storage.go

```diff
@@ -660,94 +644,123 @@ func (st *storage) storeMinerSectorsHeads(minerTips map[types.TipSetKey][]*miner
 		return xerrors.Errorf("actor put: %w", err)
 	}
 
-	if err := tx.Commit(); err != nil {
+	return tx.Commit()
+}
+
+func (st *storage) updateMinerSectors(minerTips map[types.TipSetKey][]*minerStateInfo, api api.FullNode) error {
+	log.Infow("updating miners constant sector table", "#tipsets", len(minerTips))
+	pred := state.NewStatePredicates(api)
+
+	updateTx, err := st.db.Begin()
+	if err != nil {
 		return err
 	}
-	return st.updateMinerSectors(updateMiners, api)
-}
 
-type deletedSector struct {
-	deletedSector miner_spec.SectorOnChainInfo
-	miner         address.Address
-	tskey         types.TipSetKey
-}
+	updateStmt, err := updateTx.Prepare(`UPDATE miner_sectors SET termination_epoch=$1, expiration_epoch=$2 WHERE miner_id=$3 AND sector_id=$4`)
```

Comment:
Note to self, these should be `?` instead of `$[0-9]`

Reply:
No? I thought that postgres only worked with $[0-9]``

Reply:
I'm pretty sure it's specific to the abstraction and not the database. I believe either works, where `$[0-9]` is for specifying the arg position explicitly and `?` if args should be sequentially applied.